modules/cli/src/main/scala/scala/cli/commands/SetupIde.scala

```diff
@@ -99,9 +108,18 @@ object SetupIde extends ScalaCommand[SetupIdeOptions] {
     val (bspName, bspJsonDestination) = bspDetails(inputs.workspace, options.bspFile)
     val scalaCliBspJsonDestination =
       inputs.workspace / Constants.workspaceDirName / "ide-options-v2.json"
+    val scalaCliBspInputsJsonDestination =
+      inputs.workspace / Constants.workspaceDirName / "ide-inputs.json"
 
     val inputArgs = inputs.elements.collect { case d: Inputs.OnDisk => d.path.toString }
 
+    val ideInputs = IdeInputs(args =
+      options.shared.validateInputArgs(args)
```

Comment:
It isn't a lot of overhead, but the `args` are effectively validated twice here (once for the `IdeInputs` instance, and once for the `Inputs` instance). I intend to try and refactor this in a future PR.